src/Model/Resolver/Product/PriceRange.php

```diff
@@ -32,13 +33,19 @@ class PriceRange extends CorePriceRange
      */
     protected $priceProviderPool;
 
+    /**
+     * @var PriceCurrencyInterface
+     */
+    private PriceCurrencyInterface $priceCurrency;
```

Comment:
Please don't use property types, since php 7.3 doesn't support it and it makes deploy to fail on some projects that haven't updated their php.

Reply:
I believe we have many places which use this syntax, I recall getting this error even on some recent M2 classes.